api/discord.ts

```diff
@@ -0,0 +1,81 @@
+import { URLSearchParams } from 'url';
+
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+import { request } from 'undici';
+
+type AccessTokenResponse = {
+  access_token: string;
+  expires_in: number;
+  refresh_token: string;
+  scope: string;
+  token_type: 'Bearer';
+};
+
+type DiscordMember = {
+  id: string;
+  username: string;
+  discriminator: string;
+};
+
+/**
+ * Authorization Code Grant for Discord
+ *
+ * Retrieves the discord user's access token and exchanges it for a user's access token. Uses the access token
+ * to fetch and return discord user's details, including the snowflake id.
+ *
+ * https://discord.com/developers/docs/topics/oauth2#authorization-code-grant
+ */
+export default async function handler(req: VercelRequest, res: VercelResponse) {
+  const { code } = req.query;
+
+  if (!code) {
+    console.error('code is required');
+  }
+
+  try {
+    const tokenResponseData = await request(
+      'https://discord.com/api/oauth2/token',
+      {
+        method: 'POST',
+        body: new URLSearchParams({
+          client_id: process.env.DISCORD_BOT_CLIENT_ID,
+          client_secret: process.env.DISCORD_BOT_CLIENT_SECRET,
+          code,
+          grant_type: 'authorization_code',
+          redirect_uri: process.env.DISCORD_BOT_REDIRECT_URI,
```

Comment:
one last request before merging this. Please move these env vars to the [config file](https://github.com/karelianpie/coordinape/blob/e68b6a9b018036b13dfecb59409c16b43d3ca7e9/api-lib/config.ts) and parse them with fallbacks to minimize friction for devs. For the `DICORD_BOT_CLIENT*` fallbacks, just use something like `'no_token'` and for the `DISCORD_BOT_REDIRECT_URI` add the local redirect as instructed in the test plan.

Reply:
Nice! Wasn't aware of this `config.ts`. Refactored with  https://github.com/coordinape/coordinape/pull/1625/commits/cd2de069a1c671dc52f1c84ba1bfc7d8c0425204